### PR TITLE
Fix LSTM performance on TF Backend

### LINKED ISSUE #18854: Execution time in model.fit
i got issue when i try to train my model using keras 3.0. the steps per epoch take really long time when i run it on keras 3.0 compared to when i'm using keras_core. Is there anyway to fix it ?
Here is the screenshot on keras 3.0 
image

and this one on keras-core
image

layers used in this model are : 
```py
def create_model():
    model = keras.Sequential([
        Input((778, 1)),
        Conv1D(96, 1, activation = 'relu'),
        Conv1D(352, 1,  activation = 'relu'),
        Conv1D(224, 1,  activation = 'relu'),
        Conv1D(160, 1,  activation = 'relu'),
        Conv1D(320, 1,  activation = 'relu'),
        Bidirectional(LSTM(192, return_sequences = True, activation = 'tanh')),
        Bidirectional(LSTM(384, return_sequences = True, activation = 'tanh')),
        Bidirectional(LSTM(160, return_sequences = True, activation = 'tanh')),
        Dropout(0.2),


        Dense(4, activation = 'softmax')
    ])
    model.compile(loss = 'sparse_categorical_crossentropy', optimizer = 'adam', metrics = ['accuracy'])
    return model
```
i run the model on google collab. is there any way to fix it ? Thanks

### PR DESCRIPTION
**Pull request by sampathweb**
Regression caused by #18610 
Fixes #18854 .  Also verified it doesn't re-introduce the masking issue - #18397

Wrapping check of `inputs_supported` in `tf.function` takes 200+ ms on each call.  Without this check LSTM layer returns results in 15ms/step for imdb keras.io example.  If we don't wrap it in `tf.function`, that causes masking issue since we are comparing python bool from `cudnn_supported` with TF symbolic tensor from `inputs_supported`.

This check on `inputs_supported` may not be necessary?

Colab testing the fix: https://colab.research.google.com/drive/1EFFINfOGp-htUvLZu_oK5aCzOWFNwTfS?usp=sharing

### CODE REVIEW

File: keras/backend/tensorflow/rnn.py [hunk 1]
```diff
@@ -457,15 +456,14 @@ def gru(
...
-    inputs_supported = _do_rnn_inputs_support_cudnn(mask, time_major)
     cudnn_supported = cudnn_ok(
         activation,
         recurrent_activation,
         unroll,
         use_bias=bias is not None,
         reset_after=reset_after,
     )
-    if not cudnn_supported or not inputs_supported:
+    if not cudnn_supported:
```
Comment (fchollet) [inline keras/backend/tensorflow/rnn.py:466]: Let's do the safe thing and fallback to non-cudnn if there is any mask. Do `if not cudnn_supported or mask is not None:`

**Review by fchollet — APPROVED** [review]
LGTM

File: keras/backend/tensorflow/rnn.py (hunk 1 above)
Comment (sampathweb) [reply keras/backend/tensorflow/rnn.py]: Thanks. Done.  Also verified no regression in Colab test for both issues.
